drivers/hsc/hsc.c

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (C) 2020 Deutsches Zentrum für Luft- und Raumfahrt e.V. (DLR)
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_hsc
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for the Honeywell HSC
+ *              temperature and pressure sensor.
+ *
+ * @author      Quang Pham <phhr_quang@live.com>
+ *
+ * @}
+ */
+
+#include "hsc.h"
+#include "errno.h"
+#include "hsc_internals.h"
+#include "hsc_params.h"
+#include "periph/i2c.h"
+
+#define ENABLE_DEBUG 0
+#include "debug.h"
+
+#define DEV_I2C      (dev->params.i2c_dev)
+#define DEV_ADDR     (dev->params.i2c_addr)
+
+/* Internal function prototypes */
+static int _read_ut(const hsc_t *dev, int32_t *ut);
+static int _read_up(const hsc_t *dev, int32_t *up);
+
+/*---------------------------------------------------------------------------*
+ *                          HSC Core API                                     *
+ *---------------------------------------------------------------------------*/
+
+int hsc_init(hsc_t *dev, const hsc_params_t *params)
+{
+    dev->params = *params;
+    uint8_t buf[HSC_FULL_DATA_LENGTH];
+
+    /* Acquire exclusive access */
+    if (i2c_acquire(DEV_I2C)) {
+        return -ENODEV;
+    }
+
+    if (i2c_read_bytes(DEV_I2C, DEV_ADDR, buf, sizeof(buf), 0) < 0) {
+        i2c_release(DEV_I2C);
+        DEBUG("[hsc] read_bytes fails\n");
+
+        return -EIO;
+    }
+
+    i2c_release(DEV_I2C);
+
+    /* Check the status of the sensor */
+    uint8_t status = buf[0];
+
+    switch (status & HSC_STATUS_MASK) {
+    case HSC_STATUS_OK:
+        return 0;
+    case HSC_STATUS_STALE_DATA:
+        return -EAGAIN;
+    default:
+        return -EIO;
+    }
+
+    return 0;
+}
+
+int16_t hsc_read_temperature(const hsc_t *dev)
+{
+    int32_t ut = 0;
+
+    /* Read uncompensated temperature value */
+    _read_ut(dev, &ut);
+
+    /* Formular from section 4.0 in 
+     * https://sensing.honeywell.com/i2c-comms-digital-output-pressure-sensors-tn-008201-3-en-final-30may12.pdf */
+    return (int16_t)(ut * 2000 / 2047 - 500);
+}
+
+int32_t hsc_read_pressure(const hsc_t *dev)
+{
+    int32_t up = 0;
+
+    /* Read uncompensated pressure value */
+    _read_up(dev, &up);
+
+    /* Formular from section 3.0 in 
+     * https://sensing.honeywell.com/i2c-comms-digital-output-pressure-sensors-tn-008201-3-en-final-30may12.pdf */
+    return (int32_t)((up - 1638) * 2000 * HSC_PARAM_RANGE / \
+                     (14745 - 1638) - 1000 * HSC_PARAM_RANGE);
```

Comment:
```suggestion
    return (up - 1638) * 2000 * HSC_PARAM_RANGE
           / (14745 - 1638) - 1000 * HSC_PARAM_RANGE);
```

In C a statement ends with a semicolon (;), not with the end of line. No need to escape the line break with the backslash (\). (A backslash is only needed for multi-line preprocessor statements.)

Also: As `up` is already `int32_t`, all the other values will be propagated to `int32_t` and the result will already be `int32_t`. So no need to cast here. In general: Explicit casts should be avoided when possible, as they disable type safe.